agent/fw_slim.c

```diff
@@ -97,6 +98,59 @@ NR_PHP_WRAPPER(nr_slim3_4_route_run) {
 }
 NR_PHP_WRAPPER_END
 
+static char* nr_slim_version(zval* app TSRMLS_DC) {
+  char* retval = NULL;
+  zval* version = NULL;
+  zend_class_entry* ce = NULL;
+
+  if (0 == nr_php_is_zval_valid_object(app)) {
+    nrl_verbosedebug(NRL_FRAMEWORK, "%s: Application object is invalid",
+                     __func__);
+    return NULL;
+  }
+
+  ce = Z_OBJCE_P(app);
+  if (NULL == ce) {
+    nrl_verbosedebug(NRL_FRAMEWORK, "%s: Application has NULL class entry",
+                     __func__);
+    return NULL;
+  }
+
+  version = nr_php_get_class_constant(ce, "VERSION");
+  if (NULL == version) {
+    nrl_verbosedebug(NRL_FRAMEWORK, "%s: Application does not have VERSION",
+                     __func__);
+    return NULL;
+  }
+
+  if (nr_php_is_zval_valid_string(version)) {
+    retval = nr_strndup(Z_STRVAL_P(version), Z_STRLEN_P(version));
+  } else {
+    nrl_verbosedebug(NRL_FRAMEWORK,
+                     "%s: expected VERSION be a valid string, got type %d",
+                     __func__, Z_TYPE_P(version));
+  }
+
+  nr_php_zval_free(&version);
+  return retval;
+}
+
+NR_PHP_WRAPPER(nr_slim_application_construct) {
+  zval* this_var = nr_php_scope_get(NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
+  char* version = NULL;
+
+  NR_UNUSED_SPECIALFN;
+  (void)wraprec;
+
+  version = nr_slim_version(this_var TSRMLS_CC);
+  // Add php package to transaction
```

Comment:
I think this needs to check if `NULL == version` and if so set it to the value to represent no version information.  If the version check fails it is still possible to register the slim package with no version info.  Currently if `nr_txn_add_php_package()` is called with a `NULL` version it just returns and adds nothing.

Reply:
This case is already covered in the `nr_php_packages_create()` function because when `NULL` is passed in as the version, the function will set the version to an empty string: https://github.com/newrelic/newrelic-php-agent/blob/65ca1de83400be2c7bc6bc0dbce28b1ff56ecd1d/axiom/nr_php_packages.c#L42

Reply:
Understood - I think the `nr_php_packages_create()` function should use the new `PHP_PACKAGE_VERSION_UNKNOWN` macro instead of the `" "` string constant.

Reply:
Good idea, fixed: [7687a3e](https://github.com/newrelic/newrelic-php-agent/pull/775/commits/7687a3ee65b806ff96a416452a5f8429195bcc82)